helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -336,7 +342,32 @@ public Integer apply(final Response reply) {
   }
 
   public ListenableFuture<List<String>> listHosts() {
-    return get(uri("/hosts/"), new TypeReference<List<String>>() {
+    return listHosts(ImmutableMultimap.<String, String>of());
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern) {
+    return listHosts(ImmutableMultimap.of("namePattern", namePattern));
+  }
+
+  public ListenableFuture<List<String>> listHosts(final List<String> labels) {
+    final Multimap<String, String> query = HashMultimap.create();
+    query.putAll("labels", labels);
+
+    return listHosts(query);
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern,
+                                                  final List<String> labels) {
```

Comment:
I am a-okay with `List<String>`. Functionality-wise it makes sense. It just makes the code a tad harder to follow.

Using unparsed host selectors is also a lot more friendly to the user if you do a request with `curl`.